app/models/manageiq/providers/ibm_power_hmc/infra_manager/vm.rb

```diff
@@ -96,4 +101,21 @@ def cpu_mhz_available?
   def cpu_percent_available?
     true
   end
+
+  private
+
+  def modify_attrs(attrs = {})
+    ext_management_system.with_provider_connection do |connection|
+      connection.modify_object do
+        provider_object(connection).tap do |obj|
+          attrs.each do |key, value|
+            obj.send("#{key}=", value)
+          end
+        end
```

Comment:
This is fine for now but will only work for top-level attributes.  I think longer term it would be simpler to just yield the provider_object to the caller and let it set what it wants:
```suggestion
        yield provider_object(connection)
```

```
modify_attrs { |vm| vm.name = new_name }
```

Reply:
The provider object is the SDK object. The way it works in the HMC SDK is that we define `=` accessors that modify the correct elements in the XML. This is not necessarily a top-level attribute.

Reply:
What if you wanted to set `vm.foo.bar=something` so a nested attribute?  There might not be any right now but that doesn't mean there won't be in the future

Reply:
The reason I implemented this generic method is that I want to reuse it for `raw_set_description` which is very similar to `raw_rename` except it will change the `description` attribute.